src/overlays/actors/ovl_En_Syateki_Okuta/z_en_syateki_okuta.c

```diff
@@ -35,72 +41,482 @@ const ActorInit En_Syateki_Okuta_InitVars = {
+                sp84.y = Rand_ZeroOne() * 7.0f;
+                sp84.z = (Rand_ZeroOne() - 0.5f) * 7.0f;
+                EffectSsDtBubble_SpawnCustomColor(globalCtx, &this->actor.world.pos, &sp84, &D_80A37B98, &D_80A37BA4,
+                                                  &D_80A37BA8, Rand_S16Offset(100, 50), 25, 0);
+            }
+
+            func_80A362F8(this);
+        }
+
+        this->actor.scale.y = this->actor.scale.x;
+        this->actor.scale.z = this->actor.scale.x;
+    }
+}
+
+void func_80A368E0(EnSyatekiOkuta* this, GlobalContext* globalCtx) {
+    this->collider.dim.height =
+        (sCylinderInit.dim.height - this->collider.dim.yShift) * this->unk_1D8.y * this->actor.scale.y * 100.0f;
+    this->collider.dim.radius = sCylinderInit.dim.radius * this->actor.scale.x * 100.0f;
+
+    if (this->actionFunc == func_80A363B4) {
+        if ((this->unk_2A6 == 2) && func_80A361F4(this)) {
+            return;
+        }
+
+        if (this->skelAnime.curFrame < (this->skelAnime.endFrame - 5.0f)) {
+            this->collider.dim.height *= 1.35f;
+        }
+    }
+
+    if (this->unk_2A6 == 1) {
+        this->collider.dim.radius += 10;
+        this->collider.dim.height += 15;
+    }
+
+    this->collider.dim.pos.x = this->actor.world.pos.x;
+    // jointTable->y is the y-translation of the skeleton root
+    this->collider.dim.pos.y = this->actor.world.pos.y + (this->skelAnime.jointTable->y * this->actor.scale.y);
+    this->collider.dim.pos.z = this->actor.world.pos.z;
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+}
+
+s32 func_80A36A90(EnSyatekiOkuta* this, GlobalContext* globalCtx) {
+    if ((this->actionFunc == func_80A365EC) || (this->actionFunc == func_80A36350)) {
+        return false;
+    }
+
+    if ((this->collider.base.acFlags & AC_HIT) != 0) {
+        this->collider.base.acFlags &= ~AC_HIT;
+        return true;
+    }
+
+    func_80A368E0(this, globalCtx);
+    return false;
+}
+
+void func_80A36AF8(EnSyatekiOkuta* this, GlobalContext* globalCtx) {
+    EnSyatekiMan* syatekiMan = (EnSyatekiMan*)this->actor.parent;
+    s16 temp_v1_2;
+
+    if ((this->actionFunc != func_80A36488) && (this->actionFunc != func_80A363B4) && (syatekiMan->unk_26A == 1) &&
+        (syatekiMan->unk_26C == 0)) {
+        temp_v1_2 = (syatekiMan->unk_190 >> (EN_SYATEKI_OKUTA_GET_F(&this->actor) * 2)) & 3;
+        if (temp_v1_2 > 0) {
+            Actor_SetScale(&this->actor, 0.01f);
+            this->unk_2A6 = temp_v1_2;
+            func_80A36360(this);
+        }
+    }
+}
+
+void EnSyatekiOkuta_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnSyatekiOkuta* this = THIS;
+    EnSyatekiMan* syatekiMan;
+
+    this->actionFunc(this, globalCtx);
+
+    if (this->actionFunc != func_80A36350) {
+        SkelAnime_Update(&this->skelAnime);
+    }
+
+    func_80A36AF8(this, globalCtx);
+
+    if (func_80A36A90(this, globalCtx) != 0) {
```

Comment:
```suggestion
    if (func_80A36A90(this, globalCtx)) {
```
looks to return a bool